tests/mock_env.py

```diff
@@ -0,0 +1,210 @@
+"""
+
+from __future__ import unicode_literals, absolute_import
+
+from flexmock import flexmock
+
+from atomic_reactor.constants import PLUGIN_BUILD_ORCHESTRATE_KEY
+from atomic_reactor.plugin import (PreBuildPluginsRunner,
+                                   BuildStepPluginsRunner,
+                                   PostBuildPluginsRunner,
+                                   PrePublishPluginsRunner,
+                                   ExitPluginsRunner)
+from atomic_reactor.inner import DockerBuildWorkflow
+from atomic_reactor.plugins.pre_reactor_config import (ReactorConfig,
+                                                       ReactorConfigPlugin,
+                                                       WORKSPACE_CONF_KEY)
+
+from tests.constants import TEST_IMAGE, MOCK_SOURCE
+from tests.stubs import StubSource, StubInsideBuilder
+
+
+class MockEnv(object):
+    """
+    Mock environment for unit tests.
+
+    Provides methods for setting up workflow (DockerBuildWorkflow) and runner (PluginsRunner)
+    for a specific test scenario.
+
+    Example usage:
+    >>> runner = (MockEnv()
+    >>>           .for_plugin('prebuild', 'my_plugin')
+    >>>           .set_scratch(True)
+    >>>           .make_orchestrator()
+    >>>           .create_runner(docker_tasker))  # docker_tasker is a fixture
+    >>> runner.run()
+    """
+
+    _plugin_phases = ('prebuild', 'buildstep', 'postbuild', 'prepublish', 'exit')
+
+    _runner_for_phase = {
+        'prebuild': PreBuildPluginsRunner,
+        'buildstep': BuildStepPluginsRunner,
+        'postbuild': PostBuildPluginsRunner,
+        'prepublish': PrePublishPluginsRunner,
+        'exit': ExitPluginsRunner,
+    }
+
+    _results_for_phase = {
+        'prebuild': 'prebuild_results',
+        'buildstep': 'buildstep_result',
+        'postbuild': 'postbuild_results',
+        'prepublish': 'prepub_results',
+        'exit': 'exit_results',
+    }
+
+    _plugins_for_phase = {phase: phase + '_plugins_conf' for phase in _plugin_phases}
+
+    def __init__(self):
+        self.workflow = DockerBuildWorkflow(TEST_IMAGE, source=MOCK_SOURCE)
+        self.workflow.source = StubSource()
+        self.workflow.builder = StubInsideBuilder().for_workflow(self.workflow)
+        self.workflow.builder.tasker = flexmock()
+
+        self._phase = None
+        self._plugin_key = None
+
+    def create_runner(self, docker_tasker):
+        """
+        Create runner for current plugin (configured using for_plugin())
+
+        :param docker_tasker: docker_tasker fixture from conftest
+        """
+        if self._phase is None:
+            raise ValueError('No plugin configured (use for_plugin() to configure one)')
+        runner_cls = self._runner_for_phase[self._phase]
+        plugins_conf = getattr(self.workflow, self._plugins_for_phase[self._phase])
+        return runner_cls(docker_tasker, self.workflow, plugins_conf)
+
+    def for_plugin(self, phase, plugin_key, args=None):
```

Comment:
Looks it doesn't make sense to call `for_plugin` multiple times, since the `create_runner` only creates a runner for the last phase. If this is the design, can this be documented in the docstring? Or, prevent the second call in the code and raise an error?

Reply:
Good idea, added the check and error